### DISCUSSION TITLE
Question about using FINUFFT to Fourier Transform 2D time series

### DISCUSSION THREAD

**Discussion by iamlll** [flatironinstitute/finufft#301] [FINUFFT in applications, mathematical definitions]
Hello,
This is probably a very basic question but I wasn't able to find an answer to it while reading through the FINUFFT docs, so I thought I'd ask here! I have a set of (x,y,t) particle time series that I'm interested in Fourier transforming. These trajectories are spatially non-uniform so I thought about using finufft3d1, but then saw that the coordinate inputs must be within the box [-3pi, 3pi]^3. Would you recommend rescaling the length and time scales such that all (x,y,t) coordinates lie within this box, or would it be more apt to instead use finufft3d3 for this situation?

**Comment by ahbarnett** [comment]
I'd need to understand your downstream goal to help. There are many questions that are left unclear by your post:
* Is your data a set of M>1 triplets (x_1,y_1,t_1) .... (x_M, y_M, t_M)?
* are your t values uniform (eg t_j = hj for some spacing h)?
    - if so, you could do separate FFT of x_1...x_M  and of y_1,...y_M  but you don't need NUFFT for that.
    - or, perhaps you're trying to discard t and then do a 2D NUFFT of the points (x_1,y_1), ... (x_M,y_M) ? But then the ordering in time is discarded. Since you ask about 3D NUFFT it can't be that either...
* If t_j are nonuniform, perhaps you want 1d type 3 transforms for each of the two coords, ie (x_j,t_j), then (y_j,t_j) separately?
This would only make sense if you have quadrature weights available for the t_j points.
Otherwise you need inverse NUFFT.

To answere generally the type 1 is for a Fourier *series* whereas type 3 is F *transform*.

So can you describe what the "Fourier transform" you want is more precisely? Is it for MRI?

**Comment by iamlll** [comment]
Sorry for the confusion / lack of clarity! I have discrete cell trajectories (x_i,y_i, t) with evenly spaced time steps (dt = 1) and am trying to understand the collective cell dynamics in a nonequilibrium physics context. From these trajectories, I can calculate particle displacements (ux_i(t),uy_i(t)) = (x_i-<x_i>, y_i-<y_i>). I want to Fourier transform these displacements to see if I can extract any information regarding length/timescales which are important to the dynamics of the system. 

I've tried using conventional FFT algorithms, which require that I first apply some kernel, then feed in and analyze the resulting continuous fields. However, these haven't worked so well since the data are quite noisy to begin with (and the additional kernel smoothing adds in more artifacts) and are not periodic. Since FINUFFT allows for nonuniform point pattern inputs I was wondering whether it would be a suitable choice for my datasets.

**Comment by ahbarnett** [comment]
I'm still confused about your precise goal. I encourage you to find an
example of what you want in the literature. By "kernel" you may mean
"window function" for spectral density estimation ... that is a different
issue.
Anyway, keep exploring. Best, A

On Tue, Jun 27, 2023 at 3:43 PM iamlll ***@***.***> wrote:

> Sorry for the confusion / lack of clarity! I have discrete cell
> trajectories (x_i,y_i, t) with evenly spaced time steps (dt = 1) and am
> trying to understand the collective cell dynamics in a nonequilibrium
> physics context. From these trajectories, I can calculate particle
> displacements (ux_i(t),uy_i(t)) = (x_i-<x_i>, y_i-<y_i>). I want to Fourier
> transform these displacements to see if I can extract any information
> regarding length/timescales which are important to the dynamics of the
> system.
>
> I've tried using conventional FFT algorithms, which require that I first
> apply some kernel, then feed in and analyze the resulting continuous
> fields. However, these haven't worked so well since the data are quite
> noisy to begin with (and the additional kernel smoothing adds in more
> artifacts) and are not periodic. Since FINUFFT allows for nonuniform point
> pattern inputs I was wondering whether it would be a suitable choice for my
> datasets.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flatironinstitute/finufft/discussions/301#discussioncomment-6297504>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNZRSQY3262YQ4KKWIT5TLXNMZWPANCNFSM6AAAAAAZV6ZWCQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
*-------------------------------------------------------------------~^`^~._.~'
|\  Alex Barnett    Center for Computational Mathematics, Flatiron Institute
| \                 http://users.flatironinstitute.org/~ahb     646-876-5942
